aws/internal/service/codebuild/finder/finder.go

```diff
@@ -0,0 +1,19 @@
+package finder
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codebuild"
+)
+
+// ReportGroupByArn returns the Report Group corresponding to the specified Arn.
+func ReportGroupByArn(conn *codebuild.CodeBuild, arn string) (*codebuild.BatchGetReportGroupsOutput, error) {
```

Comment:
Nit: Seems like this finder function should be doing all the work of filtering this down to `*codebuild.ReportGroup` (so `waiter.ReportGroupStatus` and other places do not need to do it).

Reply:
Will change 👍

Reply:
Done, much better now!